### PR TITLE
Create a script to deploy minikube on a VM.

### PR DESCRIPTION
**Pull request by jlewi**
* This is the first step in adding E2E testing on minikube.
* We add a function call to test_deploy that will deploy minikube on a VM.
* The script copies the relevant kubeconfig information and certificates
  to a directory.
* In a follow on PR we will incorporate this new command into our E2E
  workflow in order to create an E2E test that runs on minikube.
* Related to kubeflow/testing#6

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/459)

### CODE REVIEW

File: testing/test_deploy.py [hunk 1]
```diff
@@ -289,6 +298,122 @@ def ks_deploy(app_dir, component, params, env=None, account=None):
...
+    if content.get("error", {}).get("code") == requests.codes.CONFLICT:
+      logging.warn("VM %s already exists in zone %s in project %s ", args.vm_name, args.zone, args.project)
+    else:
+      raise
+  
+  # Locate the install minikube script.
+  install_script = os.path.join(os.path.dirname(__file__), "install_minikube.sh")
+  
+  if not os.path.exists(install_script):
+    logging.error("C %s", install_script)
```
Comment (lluunn) [inline testing/test_deploy.py]: incomplete message?

File: testing/test_deploy.py [hunk 2]
```diff
@@ -289,6 +298,122 @@ def ks_deploy(app_dir, component, params, env=None, account=None):
...
+    
+  vm_util.wait_for_vm(args.project, args.zone, args.vm_name)
+  vm_util.execute_script(args.project, args.zone, args.vm_name, install_script)
+  vm_util.execute(args.project, args.zone, args.vm_name, ["sudo minikube start --vm-driver=none --disk-size=40g"])
+    
+  # Copy the .kube and .minikube files to test_dir
```
Comment (lluunn) [inline testing/test_deploy.py]: why for loop for one item?

File: testing/test_deploy.py [hunk 3]
```diff
@@ -289,6 +298,122 @@ def ks_deploy(app_dir, component, params, env=None, account=None):
...
+    util.run(["gcloud", "compute", "--project=" + args.project, "scp",
+              "--recurse", full_target, args.test_dir, "--zone=" + args.zone])
+
+  # The .minikube directory contains some really large ISO and other files that we don't need; so we
+  # only copy the files we need.
+  minikube_dir = os.path.join(args.test_dir, ".minikube")
+  if not os.path.exists(minikube_dir):
+    os.makedirs(minikube_dir)
+  
+  for target in ["~/.minikube/*.crt", "~/.minikube/client.key"]:
```
Comment (lluunn) [inline testing/test_deploy.py]: maybe move .kube here?

File: testing/test_deploy.py [hunk 4]
```diff
@@ -289,6 +298,122 @@ def ks_deploy(app_dir, component, params, env=None, account=None):
     apply_command.append("--as=" + account)
   util.run(apply_command, cwd=app_dir)
 
+def modify_minikube_config(config_path, certs_dir):
```
Comment (lluunn) [inline testing/test_deploy.py]: what's the purpose of doing this?

Comment (jlewi) [reply testing/test_deploy.py]: We need to make the config work inside the pod and not the VM where minikube is running.
Added a comment.

File: testing/test_deploy.py (hunk 2 above)
Comment (jlewi) [reply testing/test_deploy.py]: Fixed.

File: testing/test_deploy.py (hunk 3 above)
Comment (jlewi) [reply testing/test_deploy.py]: Done.

**Comment by jlewi** [comment]
@lluunn PTAL.

File: testing/test_deploy.py (hunk 4 above)
Comment (lluunn) [reply testing/test_deploy.py]: Thanks!

**Comment by lluunn** [comment]
/lgtm
Please fix the conflict

**Pull request merged**